client/history/client.go

```diff
@@ -1053,21 +1054,34 @@ func (c *clientImpl) GetReplicationStatus(
 	request *historyservice.GetReplicationStatusRequest,
 	opts ...grpc.CallOption,
 ) (*historyservice.GetReplicationStatusResponse, error) {
-	// TODO: fan out this request to all history services
-	client, err := c.getClientForShardID(1)
-	var response *historyservice.GetReplicationStatusResponse
-	op := func(ctx context.Context, client historyservice.HistoryServiceClient) error {
-		var err error
-		ctx, cancel := c.createContext(ctx)
-		defer cancel()
-		response, err = client.GetReplicationStatus(ctx, request, opts...)
-		return err
+	clients, err := c.clients.GetAllClients()
+	if err != nil {
+		return nil, err
 	}
-	err = c.executeWithRedirect(ctx, client, op)
+
+	eg, ctx := errgroup.WithContext(ctx)
+	lock := sync.Mutex{}
+	result := &historyservice.GetReplicationStatusResponse{}
+
+	for _, client := range clients {
+		historyClient := client.(historyservice.HistoryServiceClient)
+		eg.Go(func() error {
+			resp, err := historyClient.GetReplicationStatus(ctx, request, opts...)
+			if err != nil {
+				return err
+			}
+			lock.Lock()
+			defer lock.Unlock()
+			result.Shards = append(result.Shards, resp.Shards...)
+			return nil
+		})
+	}
+	err = eg.Wait()
 	if err != nil {
 		return nil, err
 	}
```

Comment:
say there are 10 response / errors returned
above contains one error, will the other 9 success response be thrown away?

Reply:
error group cancels context when it sees first error.

Reply:
>error group cancels context when it sees first error.

exactly, as long as one error encountered, all other results are affected.
we should not use error group for this use case then